crawler/lookingglass_crawler/resilient_proxy_middleware.py

```
@@ -0,0 +1,82 @@
+        Apply proxy to request if available, but never fail.
+        """
+        if not self.enabled or not self.proxy_list:
+            return None
+            
+        # Try to get a random proxy
+        try:
+            proxy = random.choice(self.proxy_list)
+            if proxy:
+                request.meta['proxy'] = proxy
+                logger.debug(f"Using proxy: {proxy}")
+        except Exception as e:
```

Ruff (0.11.9)

Do not catch blind exception: `Exception`

(BLE001)